packages/thyra/core/src/interfaces/nonlinear/model_evaluator/fundamental/Thyra_ModelEvaluator.hpp

```diff
@@ -771,6 +850,16 @@ class ModelEvaluator : public ModelEvaluatorBase {
    */
   virtual RCP<const VectorSpaceBase<Scalar> > get_g_space(int j) const = 0;
 
+  /** \brief Return the dual vector space for the auxiliary response functions
+   * <tt>g(j) <: RE^n_g_j</tt>.
+   *
+   * <b>Preconditions:</b><ul>
+   * <li><tt>this->Ng() > 0</tt>
+   * <li><tt>0 <= j < this->Ng()</tt>
+   * </ul>
+   */
+  virtual RCP<const VectorSpaceBase<Scalar> > get_g_dual_space(int j) const { return this->get_g_space(j); };
```

Comment:
Should this not be per virtual and have the default implementation moved to the class `Thyra::ModelEvaluatorDefaultBase`?  This base class is supposed to be a pure virtual base class.